src/serializer/ResidualHeapVisitor.js

```diff
@@ -862,13 +880,14 @@ export class ResidualHeapVisitor {
       if (object.intrinsicName === "global") continue; // Avoid double-counting
       this.visitObjectProperty(propertyBinding);
     }
-    // Handing of ModifiedBindings
+    // Handling of ModifiedBindings
     for (let [additionalBinding, previousValue] of modifiedBindings) {
       let modifiedBinding = additionalBinding;
+      invariant(modifiedBinding.environment === this.resolveBinding(functionValue, modifiedBinding.name));
```

Comment:
This invariant fails. It seems to only happen when we start from additional functions that have been announced via --additionalFunctions or via the test runner.

Reply:
Resolved. This was due to the global environment record confusion.